pkg/kubectl/cmd/rollingupdate.go

```diff
@@ -138,8 +164,38 @@ func validateArguments(cmd *cobra.Command, filenames, args []string) error {
 	return utilerrors.NewAggregate(errors)
 }
 
-func RunRollingUpdate(f cmdutil.Factory, out io.Writer, cmd *cobra.Command, args []string, options *resource.FilenameOptions) error {
-	err := validateArguments(cmd, options.Filenames, args)
+func (o *RollingUpdateOptions) Complete(cmd *cobra.Command, args []string) error {
+	o.OldName = args[0]
+
+	o.DeploymentKey = cmdutil.GetFlagString(cmd, "deployment-label-key")
```

Comment:
You can bind these flags to struct `RollingUpdateOptions`, which will be easily unit tested with predictable default behavior. #60363 FYI.

/cc @soltysh